main.tf

```diff
@@ -312,22 +314,13 @@ resource "aws_launch_template" "default" {
 
 resource "aws_autoscaling_group" "default" {
   name_prefix = "${module.this.id}-asg"
-  dynamic "tag" {
-    for_each = module.this.tags
-    content {
-      key                 = tag.key
-      value               = tag.value
-      propagate_at_launch = true
-    }
-  }
-  launch_template {
-    id      = aws_launch_template.default.id
-    version = "$Latest"
-  }
 
-  max_size         = var.instance_count
-  min_size         = var.instance_count
-  desired_capacity = var.instance_count
+  max_size         = var.max_size
+  min_size         = var.min_size
+  desired_capacity = var.desired_capacity
+
+  # We don't care to protect from scale in by default, as we want to roll instances frequently
+  protect_from_scale_in = false
```

Comment:
This is [false by default](https://github.com/hashicorp/terraform-provider-aws/blob/cb64b2a54db44b45509affeb46a23845d1857e89/internal/service/autoscaling/group.go#L799).

What if we add a variable `protect_from_scale_in`, and set it to false in case a user wants to protect an instance, AND add the attribute [scale_in_protected_instances](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group#scale_in_protected_instances) to `aws_autoscaling_group`:
> `scale_in_protected_instances` - (Optional) Behavior when encountering instances protected from scale in are found. Available behaviors are `Refresh`, `Ignore`, and `Wait`. Default is Ignore.

Reply:
👍 Will do.